hooks/tk-multi-publish2/ingest/collector.py

```diff
@@ -0,0 +1,257 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import mimetypes
+import os
+import datetime
+import glob
+import pprint
+import urllib
+import sgtk
+from sgtk import TankError
+
+from tank import context
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class IngestCollectorPlugin(HookBaseClass):
+    """
+    Collector that operates on the current set of ingestion files. Should
+    inherit from the basic collector hook.
+    """
+
+    def _add_file_item(self, settings, parent_item, path, is_sequence=False, seq_files=None):
+        """
+        Creates a file item
+
+        :param dict settings: Configured settings for this collector
+        :param parent_item: parent item instance
+        :param path: Path to analyze
+        :param is_sequence: Bool as to whether to treat the path as a part of a sequence
+        :param seq_files: A list of files in the sequence
+
+        :returns: The item that was created
+        """
+        publisher = self.parent
+
+        # get info for the extension
+        item_info = self._get_item_info(settings, path, is_sequence)
+
+        icon_path = item_info["icon_path"]
+        item_type = item_info["item_type"]
+        type_display = item_info["type_display"]
+        work_path_template = item_info["work_path_template"]
+
+        display_name = publisher.util.get_publish_name(path)
+
+        # Define the item's properties
+        properties = {}
+
+        # set the path and is_sequence properties for the plugins to use
+        properties["path"] = path
+        properties["is_sequence"] = is_sequence
+
+        # If a sequence, add the sequence path
+        if is_sequence:
+            properties["sequence_paths"] = seq_files
+
+        if work_path_template:
+            properties["work_path_template"] = work_path_template
+
+        # build the context of the item
+        context = self._get_item_context_from_path(parent_item, properties, os.path.basename(path))
```

Comment:
Rather than overriding _add_file_item(), why don't you just override _get_item_context_from_path() and _resolve_work_path_template() to run the os.path.basename() and then call the parent superclass?

Reply:
Makes sense, making the change.